newlib/libc/sys/vita/fs.c

```diff
@@ -43,10 +49,480 @@ int mkdir(const char *pathname, mode_t mode)
 int rmdir(const char *pathname)
 {
 	int ret;
-	if ((ret = sceIoRmdir(pathname)) < 0) {
-		errno = ret & SCE_ERRNO_MASK;
+	char* full_path = __realpath(pathname);
+	if(!full_path)
+	{
+		errno = errno; // set by realpath
 		return -1;
 	}
+
+	if ((ret = sceIoRmdir(full_path)) < 0)
+	{
+		free(full_path);
+		errno = __vita_sce_errno_to_errno(ret, ERROR_GENERIC);
+		return -1;
+	}
+	free(full_path);
 	errno = 0;
 	return 0;
 }
+
+int lstat(const char *path, struct stat *buf)
+{
+	return _stat_r(_REENT, path, buf);
+}
+
+int chmod(const char *pathname, mode_t mode)
+{
+	struct SceIoStat stat = {0};
+	char* full_path = __realpath(pathname);
+	if(!full_path)
+	{
+		errno = errno; // set by realpath
```

Comment:
ENOENT or EIO?

Reply:
EINVAL/ENOENT/ENOMEM.Removed redundant assignments from rmdir and chmod, leaving only comment.